discovery/dockerswarm/nodes.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerswarm
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/util/strutil"
+)
+
+const (
+	swarmLabelNodePrefix               = swarmLabel + "node_"
+	swarmLabelNodeAddress              = swarmLabelNodePrefix + "address"
+	swarmLabelNodeAvailability         = swarmLabelNodePrefix + "availability"
+	swarmLabelNodeEngineVersion        = swarmLabelNodePrefix + "engine_version"
+	swarmLabelNodeHostname             = swarmLabelNodePrefix + "hostname"
+	swarmLabelNodeID                   = swarmLabelNodePrefix + "id"
+	swarmLabelNodeLabelPrefix          = swarmLabelNodePrefix + "label_"
+	swarmLabelNodeManagerAddr          = swarmLabelNodePrefix + "manager_address"
+	swarmLabelNodeManagerLeader        = swarmLabelNodePrefix + "manager_leader"
+	swarmLabelNodeManagerReachability  = swarmLabelNodePrefix + "manager_reachability"
+	swarmLabelNodePlatformArchitecture = swarmLabelNodePrefix + "platform_architecture"
+	swarmLabelNodePlatformOS           = swarmLabelNodePrefix + "platform_os"
+	swarmLabelNodeRole                 = swarmLabelNodePrefix + "role"
+	swarmLabelNodeStatus               = swarmLabelNodePrefix + "status"
+)
+
+func (d *Discovery) refreshNodes(ctx context.Context) ([]*targetgroup.Group, error) {
+	tg := &targetgroup.Group{
+		Source: "DockerSwarm",
+	}
+
+	nodes, err := d.client.NodeList(ctx, types.NodeListOptions{Filters: makeArgs(d.filters)})
+	if err != nil {
+		return nil, fmt.Errorf("error while listing swarm nodes: %w", err)
+	}
+
+	for _, n := range nodes {
+		labels := model.LabelSet{
```

Comment:
You're still using model here and a few other places.

Reply:
This is the list of targets, not an intermediate step, I have to.

Reply:
I have found some placed where that was not needed